lib/pos_mempool_nonce_tracker.go

```diff
@@ -0,0 +1,52 @@
+package lib
+
+// PosMempoolNonceTracker is a helper struct that is used to track (public key, nonce) pairs in pos mempool.
+// It is used to facilitate the "replace by higher fee" feature, which allows users to resubmit transactions.
+// If a user submits a transaction with an existing DeSoNonce.PartialID (among this user's transactions in the mempool),
+// then the new transaction will replace the old one if the new transaction has a higher fee.
+type PosMempoolNonceTracker struct {
+	// nonceMap indexes mempool transactions by (public key, nonce) pairs.
+	nonceMap map[PublicKey]map[uint64]*MempoolTx
```

Comment:
Nonces are actually PKID + Uint64. We should use PKIDs so we can continue support for swap identity transactions. Additionally, I'd recommend we use the DeSoNonce struct as the map key instead of having nested maps, unless there's a use case for looking up all the nonces for a given public key that I'm unaware of.
```suggestion
	nonceMap map[DeSoNonce]*MempoolTx
```

Reply:
This suggestion isn't correct. We can create a new struct that has a transactor PKID (or public key, still need to sort this piece out) plus the DeSoNonce so we can avoid the nested map.

Reply:
Just clarifying Nina's comments:

Nonces are made up of the tuple (ExpirationBlockHeight, PartialID). You'll need the map lookup keys to contain both of these values. As Nina suggested, the `DeSoNonce` is the right struct to use to represent the nonce, not uint64.

On using PublicKey vs PKID to identify the user, I don't have a strong opinion on it. Which every key we use, should be consistent across the whole of the mempool though. I.e. if we use PKID, then the ledge should use PKID as well.